### PR TITLE
Run functional tests on STDCI

### PR DESCRIPTION
**Pull request by gbenhaim**
Run check-patch.sh in STDCI on each PR.
check-patch.sh will create a cluster and run the functional tests.
stdci.yaml indicates that the automation should run on a bare-metal
server.

Signed-off-by: gbenhaim <galbh2@gmail.com>

### CODE REVIEW

**Comment by gbenhaim** [comment]
@phoracek If you'll rebase https://github.com/kubevirt/kubernetes-device-plugins/pull/47 on this PR, the functional tests will be checked on STDCI.

File: automation/check-patch.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash -xe
+
+install_deps() {
+    pip install pytest
+}
+
+main() {
+    install_deps
+    trap 'make cluster-down' EXIT
+    make cluster-up
```
Comment (phoracek) [inline automation/check-patch.sh]: We should do `make build` too, but for that we need to install golang.
How should we proceed? Do you want to spend time debugging it or do we just merge empty main? I can fix it later.

Comment (gbenhaim) [reply automation/check-patch.sh]: Which version of golang the build needs?

Comment (phoracek) [reply automation/check-patch.sh]: Not sure, I use 1.9.7

**Comment by gbenhaim** [comment]
- The automation will use golang-1.9.4 (this is the latest 1.9.x version available in Centos repos).
- I've added `make build`

Once the hook is ready, I'll update Jenkins and then we can test this PR.

**Comment by phoracek** [comment]
@gbenhaim  thanks, I created those two hooks:
http://jenkins.ovirt.org/ghprbhook/ (issue_comment and pull_request)
http://jenkins.ovirt.org/generic-webhook-trigger/invoke (push)

**Comment by ovirt-infra** [comment]
Some tests failed

**Comment by gbenhaim** [comment]
@phoracek Can you please give write permissions to `ovirt-infra` ?

**Comment by gbenhaim** [comment]
ci test please

**Comment by phoracek** [comment]
@gbenhaim I think we need python-devel package in dependencies

**Comment by gbenhaim** [comment]
I've added python-devel.
Any idea what should be the `GOPATH`?

**Comment by gbenhaim** [comment]
@phoracek ping

**Comment by phoracek** [comment]
@gbenhaim sorry, I'm changing it, so it will be built in containers, will post a patch for it today and we can rebase this on top of it. Should be easier.

**Comment by eedri** [comment]
@phoracek any update? we've love to get it in for Sprint 13 until Wed, 
Do you think it'll be possible?

**Comment by phoracek** [comment]
@eedri @gbenhaim I'm not sure whether I will have time to debug it until Wednesday, could we make it so it is just here with `echo hello world` and I would debug it and add tests execution later?

**Comment by eedri** [comment]
I'm OK to enable the STD-CI and continue to work on the issues once we have it running,
It looks like a challenge that will take a few days to resolve, either to run with latest Docker (not on CentOS) or to use Buildah.

**Review by phoracek — APPROVED** [review]

**Pull request merged**